addons/account/models/sequence_mixin.py

```diff
@@ -20,6 +20,36 @@ class SequenceMixin(models.AbstractModel):
     _sequence_yearly_regex = r'^(?P<prefix1>.*?)(?P<year>\d{4})(?P<prefix2>\D+?)(?P<seq>\d*)(?P<suffix>\D*?)$'
     _sequence_fixed_regex = r'^(?P<prefix1>.*?)(?P<seq>\d*)(?P<suffix>\D*?)$'
 
+    sequence_prefix = fields.Char(compute='_compute_split_sequence', store=True)
+    sequence_number = fields.Integer(compute='_compute_split_sequence', store=True)
+
+    def __init__(self, pool, cr):
+        # add the dependance on the dynamic field name `_sequence_field`
+        api.depends(self._sequence_field)(type(self)._compute_split_sequence)
+        return super().__init__(pool, cr)
+
+    def init(self):
+        # Add an index to optimise the query searching for the highest sequence number
+        if not self._abstract:
+            index_name = self._table + '_sequence_index'
+            self.env.cr.execute('SELECT indexname FROM pg_indexes WHERE indexname = %s', (index_name,))
+            if not self.env.cr.fetchone():
+                self.env.cr.execute("""
+                CREATE INDEX {index_name} ON {table}
+                    (journal_id, sequence_prefix desc, sequence_number desc, date, {field});
```

Comment:
Weird to assume that the inherited model will have the `journal_id` and `date` columns.